aeronautical/asterix/asterix_data_block.ksy

```diff
@@ -0,0 +1,35 @@
+meta:
+  id: asterix_data_block
+  file-extension: ast
+  endian: be
+  license: GPL-3.0-only
```

Comment:
Can you explain (1) what's the expected consequence of applying GPL3 to these files? (2) why did you make that decision — i.e. are there any reasons which force GPL3, or is it your personal choice?

Reply:
I want it remain open source. Kolanich suggested AGPL, I think I will switch, but have no opportunity to make it yet.

Reply:
The usual disclaimer that I'm not a lawyer applies.

The problem with GPL or AGPL is that it's very unclear what that means in relation to KSY. KSY specifications are not exactly "code" — they're more like documentation. There is no single straightforward way to translate "source code" into "object code", it's unclear what constitutes a "modification" or "modified version", what consitutes "propagation", etc.

For example, if your original intent was to make sure it's "viral", i.e. everyone who uses your specs will need to apply GPLv3 to their whole codebase, then how would you deal with:

* Somebody using reference ksc compiler to generate code based on your .ksy files?
* Somebody reading your specs as documentation and implementing their own parser code based on that?
* Somebody rendering your specs as a chart using reference ksc compiler + then implementing their own parser code looking at the resulting charts?
* Somebody using alternative compiler / totally alternative approach to generate some code based on your .ksy files?

This is pretty gray legal area.

Reply:
Is ksc a compiler?

Reply:
We usually abbreviate `Kaitai Struct Compiler` (in fact it is a transpiler) as `ksc`.

Reply:
Ksc is a compiler, it translates some source file (ksy) into some compiled code for target you want.
The Ksy files in this folder could be enhanced, I still expect this, and I want those enhanced and new ksy remain open source.
Answering your questions:
1) They have to attach those ksy files, as a source code.
2) If they implement their own I could not stop them.
3) same as above.
4) same as 1.

Reply:
2. and 3. contradicts 4